src/github.com/matrix-org/go-neb/services/feedreader/feedreader.go

```diff
@@ -0,0 +1,235 @@
+package services
+
+import (
+	"errors"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/matrix-org/go-neb/database"
+	"github.com/matrix-org/go-neb/matrix"
+	"github.com/matrix-org/go-neb/polling"
+	"github.com/matrix-org/go-neb/types"
+	"github.com/mmcdole/gofeed"
+	"html"
+	"time"
+)
+
+const minPollingIntervalSeconds = (10 * 60) // 10min
+
+type feedPoller struct{}
+
+func (p *feedPoller) IntervalSecs() int64 { return 10 }
+func (p *feedPoller) OnPoll(s types.Service, cli *matrix.Client) {
+	logger := log.WithFields(log.Fields{
+		"service_id":   s.ServiceID(),
+		"service_type": s.ServiceType(),
+	})
+
+	frService, ok := s.(*feedReaderService)
+	if !ok {
+		logger.Error("FeedReader: OnPoll called without an Feed Service instance")
+		return
+	}
+	now := time.Now().Unix() // Second resolution
+
+	// Work out which feeds should be polled
+	var pollFeeds []string
+	for u, feedInfo := range frService.Feeds {
```

Comment:
It's a Go convention to be terse with variable names (they prefer single letters or just a few).